File: clients/app/android/app/src/main/java/com/weiqi/app/util/Logger.kt
package com.weiqi.app.util

import android.util.Log
import com.weiqi.app.debug.DebugBridge

/**
 * 日志工具类
 * 
 * 统一日志输出：
 * 1. 输出到 Logcat（方便开发调试）
 * 2. 输出到 DebugBridge（方便调试页面查看）
 */
object Logger {
    
    fun d(tag: String, message: String) {
        DebugBridge.log("DEBUG", tag, message)
    }
    
    fun i(tag: String, message: String) {
        DebugBridge.log("INFO", tag, message)
    }
    
    fun w(tag: String, message: String, throwable: Throwable? = null) {
        val fullMessage = if (throwable != null) {
            "$message: ${throwable.message}"
        } else {
            message
        }
        DebugBridge.log("WARN", tag, fullMessage)
    }
    
    fun e(tag: String, message: String, throwable: Throwable? = null) {
        val fullMessage = if (throwable != null) {
            "$message: ${throwable.message}"
        } else {
            message
        }
        DebugBridge.log("ERROR", tag, fullMessage)
    }
}

File: clients/desktop/src/main/utils/logger.ts
/**
 * 日志工具
 */

export enum LogLevel {
  DEBUG = 0,
  INFO = 1,
  WARN = 2,
  ERROR = 3,
}

class Logger {
  private level: LogLevel = LogLevel.INFO;
  private context: string;

  constructor(context: string) {
    this.context = context;
  }

  setLevel(level: LogLevel) {
    this.level = level;
  }

  debug(message: string, ...args: any[]) {
    if (this.level <= LogLevel.DEBUG) {
      console.debug(`[${this.context}] [DEBUG] ${message}`, ...args);
    }
  }

  info(message: string, ...args: any[]) {
    if (this.level <= LogLevel.INFO) {
      console.log(`[${this.context}] [INFO] ${message}`, ...args);
    }
  }

  warn(message: string, ...args: any[]) {
    if (this.level <= LogLevel.WARN) {
      console.warn(`[${this.context}] [WARN] ${message}`, ...args);
    }
  }

  error(message: string, error?: any) {
    if (this.level <= LogLevel.ERROR) {
      console.error(`[${this.context}] [ERROR] ${message}`, error || '');
      if (error?.stack) {
        console.error(error.stack);
      }
    }
  }
}

export function createLogger(context: string): Logger {
  return new Logger(context);
}
